### PR TITLE
Add tests for consuming instance extension methods from VB and previous version of C# compiler.

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@jcouv, @jjonescz Please review

File: src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs [hunk 1]
```diff
@@ -42696,4 +42742,1408 @@ .maxstack 8
...
+    {
+    	.maxstack 8
+    	IL_0000: ldarg.0
+    	IL_0001: call instance void [mscorlib]System.Object::.ctor()
+    	IL_0006: ret
+    }
+}
+""";
+
+#if !PREVIOUSVERSIONOFCOMPILER
```
Comment (jjonescz) [inline src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs:42825]: What's `PREVIOUSVERSIONOFCOMPILER`? It's not defined anywhere as far as I can see.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs:42825]: > It's not defined anywhere as far as I can see.

That is the intent. The branch that is not taken reflects behavior of the compiler in `main`. Obviously, it is going to fail in `roles`. The tests can be run manually in `main' by copy/pasting them into a file and defining the conditional compilation symbol.

**Comment by AlekseyTs** [comment]
@jcouv, @jjonescz, @dotnet/roslyn-compiler Please review

**Comment by AlekseyTs** [comment]
@jcouv, @jjonescz, @dotnet/roslyn-compiler Please review

File: src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs (hunk 1 above)
Comment (jcouv) [reply src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs:42825]: Consider leaving a comment on one of those directives to provide context.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

File: src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs:42825]: > Consider leaving a comment on one of those directives to provide context.

I will do this in a separate PR

**Pull request merged**